modeling-data-relationships/modeling-answers.md

```diff
@@ -0,0 +1,121 @@
+1)
+
+a)
+
+```
+SELECT guests.first_name, guests.last_name
+FROM guests
+LEFT JOIN bookings ON bookings.guest_id = guests.id
+WHERE bookings.guest_id IS NULL;
+```
+b) 
+```
+SELECT b.check_in, b.check_out, g.first_name, g.last_name, COUNT(*)
+FROM bookings AS b
+JOIN guests AS g ON g.id = b.guest_id
+GROUP BY b.check_in, b.check_out, g.first_name, g.last_name
+HAVING COUNT(*) = 2;
+```
+c)
+```
+SELECT b.room_id, g.first_name, g.last_name, COUNT(*)
+FROM guests AS g
+JOIN bookings AS b ON b.guest_id = g.id
+GROUP BY b.room_id, g.first_name, g.last_name
+HAVING COUNT(*) > 1;
+```
+
+d)
+```
+SELECT bookings.room_id, COUNT(DISTINCT guests.first_name) AS unique_guests
+FROM bookings 
+JOIN guests ON guests.id = bookings.guest_id
+GROUP BY bookings.room_id;
+```
+2)
+
+a)
+
+**What classes/entities do we need?**
+
+We definitely need a `students` entity, as well as a `course` entity in order to track the students in the school, as well as the courses give in the school. 
+
+Since the question requires that we have also the grades for each student taking a class, we will need a `grades` entity as well. 
+
+**What fields/attributes will each entity need**
+
+A `students` entity will need the following attributes `id`, `first_name`, and `last_name`. 
+
+The `courses` entity will need the following attributes: `id`, and `course_name`. 
+
+The `grades` entity will need the following attributes: `id`, `course_id`, and `student_id`.
+
+**What data types do we need to use?**
+
+For the `students`, `id` should be an INTEGER and a PRIMARY KEY, `first_name` and `last_name` should be VARCHAR.
+
+For `courses`, `id` should be an INTEGER and a PRIMARY KEY, `course_name` should be a VARCHAR and `credits` as an INTEGER. 
+
+For `grades`, `id` should be an INTEGER and the PRIMARY KEY.`course_id` is a FOREIGN KEY that REFERENCES `courses.id`,and `student_id` is also a FOREIGN KEY that REFERENCES `students.id`. Both of these FOREIGN KEYS should be an INTEGER. Finally, grade is a CHAR(1), that represents the single letter grade.
```

Comment:
no numeric value? you shouldn't have a grade column inside a grades table

Reply:
I thought letters were only necessary since the queries in question 3) dictated the necessary columns.